compiler/src/iree/compiler/Dialect/Flow/Transforms/SplitReduction.cpp

```diff
@@ -76,14 +82,87 @@ struct LinalgSplitReduction
   LinalgExt::LinalgTransformationFilter filter;
 };
 
+/// Sets the split reduction value for topk operation using a simple heuristic.
+/// The goal here is to increase parallelism.
+static int64_t topkSplitReduceRatio(int64_t splitReductionDepth,
+                                    LinalgExt::TopkOp topkOp, int64_t kValue) {
+  // Step 0. If user set the split reduction ratio use it.
+  if (!topkSplitReductionRatio.empty()) {
+    SmallVector<int64_t, 4> reductionRatios(topkSplitReductionRatio.begin(),
+                                            topkSplitReductionRatio.end());
+    if (splitReductionDepth >= reductionRatios.size()) {
+      return kNOTopkSplitReductionRatioDefault;
+    } else {
+      return reductionRatios[splitReductionDepth];
+    }
+  }
+
+  // Step 1. Hard to predict of advantage of splitting reduction more than 2
+  // depth.
+  if (splitReductionDepth > 1) return kNOTopkSplitReductionRatioDefault;
```

Comment:
Should we use the `reductionRatios` here instead of the default?

I.e., I believe that's more in line with what the code used to do. (Could be wrong though)

Reply:
`reductionRatios` is filled by the `topkSplitReductionRatio` flag. 

I think we really want to return here `kNOTopkSplitReductionRatioDefault` (that is `-1`) to not do further split reduction.

Reply:
Sounds good.